unit-tests/src/test/scala/scala/scalanative/posix/TimeTest.scala

```diff
@@ -133,6 +185,61 @@ class TimeTest {
     }
   }
 
+  @Test def strptimeDoesNotWriteMemoryOutsideStructTm(): Unit = {
+    Zone { implicit z =>
+      // Linux 56 Bytes, Posix specifies 36 but allows more.
+      val tmBufSize = 7
+      val tmBuf     = alloc[Ptr[Byte]](tmBufSize) // will zero/clear all bytes
+      val tmPtr     = tmBuf.asInstanceOf[Ptr[tm]]
+
+      val cp =
+        strptime(c"Fri Mar 31 14:47:44 EDT 2017", c"%a %b %d %T %Z %Y", tmPtr)
+
+      assertNotNull(s"strptime() returned unexpected null pointer", cp)
+
+      val ch = cp(0)
+      assertEquals("strptime() result is not NUL terminated", ch, '\u0000')
+
+      val tm_gmtoff = tmBuf(5) // tm_gmtoff is outside posix minimal range.
+      assertNull("tm_gmtoff", null)
+
+      val tm_zone = tmBuf(6) // tm_zone is outside posix minimal range.
```

Comment:
Keep in mind than `tm_gmtoff` is `long` and not `int.

Reply:
I have re-worked that section, keeping your caution about long in mind and
considering 32 vs 64 bit longs.  Thank you.